buildbot/master/files/config/factories.py

```diff
@@ -103,6 +104,95 @@ def make_step(self, command):
         return step_class(**step_kwargs)
 
 
+class StepsYAMLParsingStep(buildstep.ShellMixin, buildstep.BuildStep):
+    """
+    Step which resolves the in-tree YAML and dynamically add test steps.
+    """
+
+    haltOnFailure = True
+    flunkOnFailure = True
+
+    def __init__(self, builder_name, environment, yaml_path):
+        self.builder_name = builder_name
+        self.environment = environment
+        self.yaml_path = yaml_path
+
+    @defer.inlineCallbacks
+    def run(self):
+        try:
+            full_yaml_path = os.path.join(self.build.workdir, self.yaml_path)
+            print_yaml_cmd = "cat {}".format(full_yaml_path)
+            cmd = yield self.makeRemoteShellCommand(command=[print_yaml_cmd],
+                                                    collectStdout=True)
+            yield self.runCommand(cmd)
+
+            if cmd.didFail():
+                raise Exception(str(cmd.result()))
```

Comment:
I don't think there's a `result` method on `RemoteShellCommand`.
It may be better to capture stderr and output that, or maybe return a message with the return code (`cmd.rc`).